app/features/receive/receive-provider.tsx

```diff
@@ -0,0 +1,43 @@
+import {
+  type PropsWithChildren,
+  createContext,
+  useContext,
+  useRef,
+} from 'react';
+import { useStore } from 'zustand';
+import type { Account } from '../accounts/account-selector';
+import type { ReceiveState, ReceiveStore } from './receive-store';
+import { createReceiveStore } from './receive-store';
+
+const ReceiveContext = createContext<ReceiveStore | null>(null);
+
+type Props = PropsWithChildren<{
+  /** Usually the user's default account. This sets the initial account to receive to. */
+  initialAccount: Account;
+}>;
+
+export const ReceiveProvider = ({ children, initialAccount }: Props) => {
+  const storeRef = useRef<ReceiveStore>();
```

Comment:
I think in this case you can also use state hook instead:
```
const [store] = useState<ReceiveStore>(() => reateReceiveStore({
      initialAccount,
      initialAmount: null,
}));
```

it's a bit more succinct